### PR TITLE
feat: move general functionality to the outside package #1

### LINKED ISSUE #1: Move general functions from the main app
Move utility functions, Flow interaction functions, interfaces and types.

### PR DESCRIPTION
**Pull request by iliawx**
Closes #1

### CODE REVIEW

File: lib/type/i-flow-purchase-event.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface IFlowPurchaseEvent {
```
Comment (sampullman) [inline lib/type/i-flow-purchase-event.ts:1]: This one is app-specific and shouldn't be included

File: lib/util/parse-cadence-events.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { IFlowPurchaseEvent } from '../type/i-flow-purchase-event'
```
Comment (sampullman) [inline lib/util/parse-cadence-events.spec.ts:1]: App-specific and shouldn't be included

File: lib/util/parse-cadence-events.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { CadenceEvent } from '@samatech/onflow-fcl-esm'
```
Comment (sampullman) [inline lib/util/parse-cadence-events.ts:1]: App-specific and shouldn't be included

**Review by sampullman — CHANGES_REQUESTED** [review]
Couple of notes on app-specific things that shouldn't be included, otherwise looks pretty good!

File: lib/type/i-flow-purchase-event.ts (hunk 1 above)
Comment (iliawx) [reply lib/type/i-flow-purchase-event.ts]: Removed

File: lib/util/parse-cadence-events.spec.ts (hunk 2 above)
Comment (iliawx) [reply lib/util/parse-cadence-events.spec.ts]: Removed

File: lib/util/parse-cadence-events.ts (hunk 3 above)
Comment (iliawx) [reply lib/util/parse-cadence-events.ts]: I'm guessing you've meant `IFlowPurchaseEvent` here as well. I removed it.

**Review by iliawx — COMMENTED** [review]
I've removed the app-specific `IFlowPurchaseEvent` interface and consolidated misc flow functions.

**Pull request merged**